share/spice/timer/timer.js

```diff
@@ -406,19 +440,31 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
     });
 
     env.ddg_spice_timer = function(api_result) {
+        var timers = [],
+            timerInterval,
+            $addTimerBtn;
+
+        function addTimer(startingTime) {
+            var timer = new Timer(timers.length + 1, startingTime);
+            timer.$element.insertBefore($addTimerBtn.parent());
+            timers.push(timer);
+        }
         
         function onShow() {
```

Comment:
:+1: We implemented it this way to provide a lower level of control over things from Spice. We could add abstractions on top, like `onFirstShow` or something.